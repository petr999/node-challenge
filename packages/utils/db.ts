import { Client } from 'pg';
import config from 'config';
import { Connection } from 'typeorm';

export const dbConnection: {
  typeOrm?: Connection
  [k: string]: any
} = {};

export function connect() {
  dbConnection.client = new Client(config.db);
  return dbConnection.client.connect();
}

export async function query(queryString: string, parameters?: any) {
  try {
    if (!dbConnection.client) await connect();
  } catch (e) {
    dbConnection.client = undefined;
    throw e;
  }

  return dbConnection.typeOrm.query(queryString, parameters);
}
