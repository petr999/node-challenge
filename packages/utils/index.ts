export * from './formatter';
export * from './dal';
export * from './errors';
