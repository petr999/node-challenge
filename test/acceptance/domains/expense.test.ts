import { Api } from '../utils/api';

describe('Unknown user\'s expenses not found', () => {
  test('Random user returns Not Found', (done) => {
    Api.get('/expense/v1/user/12992732-bd00-4a9a-9426-131bd618740e')
      .expect(404, done);
  });
});

describe('User expense found in database', () => {
  test('User expense known to be contained in database got fetched', (done) => {
    Api.get('/expense/v1/user/da140a29-ae80-4f0e-a62d-6c2d2bc8a474')
      .expect(200)
      .end((err, res) => {
        expect(err).toBeFalsy();
        expect(res.body).toMatchSnapshot('expense-user-by-user-id-da140a29-ae80-4f0e-a62d-6c2d2bc8a474');
        done();
      });
  });
});
