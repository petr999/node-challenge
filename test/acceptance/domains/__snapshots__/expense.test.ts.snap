// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User expense found in database User expense known to be contained in database got fetched: expense-user-by-user-id-da140a29-ae80-4f0e-a62d-6c2d2bc8a474 1`] = `"[[{\\"id\\":\\"3e920f54-49df-4d0b-b11b-e6f08e3a2dca\\",\\"dateCreated\\":\\"2021-09-21T20:57:40.021Z\\",\\"userId\\":\\"da140a29-ae80-4f0e-a62d-6c2d2bc8a474\\",\\"merchantName\\":\\"Cafe 22\\",\\"currency\\":\\"DKK\\",\\"amount\\":\\"80.00\\"},{\\"id\\":\\"314d54f4-8a5f-4c1d-b735-426b54794a44\\",\\"dateCreated\\":\\"2021-09-20T20:57:40.021Z\\",\\"userId\\":\\"da140a29-ae80-4f0e-a62d-6c2d2bc8a474\\",\\"merchantName\\":\\"Sliders\\",\\"currency\\":\\"DKK\\",\\"amount\\":\\"120.00\\"},{\\"id\\":\\"f20866f9-7d46-45f2-822c-4b568e216a13\\",\\"dateCreated\\":\\"2021-09-19T20:57:40.021Z\\",\\"userId\\":\\"da140a29-ae80-4f0e-a62d-6c2d2bc8a474\\",\\"merchantName\\":\\"Donkey Republic\\",\\"currency\\":\\"DKK\\",\\"amount\\":\\"60.00\\"}],3]"`;
